internal/client/client_interceptor.go

```diff
@@ -107,6 +107,8 @@ func buildResourceControlInterceptor(
 	// Build the interceptor.
 	interceptFn := func(next interceptor.RPCInterceptorFunc) interceptor.RPCInterceptorFunc {
 		return func(target string, req *tikvrpc.Request) (*tikvrpc.Response, error) {
+			// Background requests don't need to update the runtime stats because it will record in TiKV.
+			reqInfo.SetRequestSource(util.RequestSourceFromCtx(ctx))
```

Comment:
I think the context here in the client-go is not trustable, you can direct get from `tikvrpc.Reqeust`. in `MakeRequestInfo` is parsed different requests.